src/msca/metrics/main.py

```diff
@@ -0,0 +1,249 @@
+from enum import StrEnum, auto
+import numpy as np
+import pandas as pd
+
+from sklearn import metrics
+
+
+class Metric(StrEnum):
+    """
+    A metric enum that can be instantiated with string names and supports both error and skill calculations.
+
+    Examples
+    --------
+    >>> # Simple usage
+    >>> metric = Metric("mean_absolute_error")
+    >>> score = metric.eval(df, "obs", "pred", "weights")
+    >>> # Grouped calculation
+    >>> grouped_scores = metric.eval(
+    ...     df, "obs", "pred", "weights", groupby=["region"]
+    ... )
+    >>>
+    >>> # Skill calculation
+    >>> skill_score = metric.eval(
+    ...     df, "obs", "pred_ref", "weights", pred_alt="pred_alt"
+    ... )
+    """
+
+    MEAN_ABSOLUTE_ERROR = auto()
+    MEAN_ABSOLUTE_PERCENTAGE_ERROR = auto()
+    MEAN_SQUARED_ERROR = auto()
+    MEDIAN_ABSOLUTE_ERROR = auto()
+    ROOT_MEAN_SQUARED_ERROR = auto()
+
+    def eval(
+        self,
+        data: pd.DataFrame,
+        obs: str,
+        pred_ref: str,
+        weights: str,
+        pred_alt: str | None = None,
+        groupby: list[str] | None = None,
+    ) -> float | pd.DataFrame:
+        """
+        Evaluate the error metric or skill score on the provided data.
```

Comment:
I think we should have two functions, `eval` and `eval_skill` to keep them more manageable.